File: frontend/src/types/statistics.ts
/**
 * 统计相关类型定义
 */

export interface StatisticsOverview {
  total_messages: number
  today_messages: number
  total_links: number
}

export interface DailyTrendItem {
  date: string
  messages: number
  links: number
}

export interface DailyTrendResponse {
  days: DailyTrendItem[]
}

export interface DedupStatsItem {
  hour: string
  deleted_count: number
}

export interface DedupStatsResponse {
  hours: DedupStatsItem[]
}

export interface NetdiskDistributionItem {
  netdisk_name: string
  link_count: number
  percentage: number
}

export interface NetdiskDistributionResponse {
  distribution: NetdiskDistributionItem[]
}

export interface ActivityHeatmapCell {
  date: string
  hour: number
  message_count: number
}

export interface ActivityHeatmapResponse {
  dates: string[]
  hours: number[]
  cells: ActivityHeatmapCell[]
  max_count: number
}

export interface AdminChannelMatrixRow {
  row_key: string
  monitor_channel_config_id?: number | null
  monitor_channel_key?: string | null
  monitor_channel_title: string
  total_messages: number
  total_links: number
  message_counts: Record<string, number>
  link_counts: Record<string, number>
  trend: number[]
}

export interface AdminChannelMatrixResponse {
  days: number
  dates: string[]
  rows: AdminChannelMatrixRow[]
  available_since?: string | null
  max_daily_messages: number
}

File: app/schemas/statistics.py
"""
统计相关的 Pydantic Schema
"""

from pydantic import BaseModel
from typing import List, Dict, Any, Optional
from datetime import datetime


class StatisticsOverview(BaseModel):
    """总体统计"""
    total_messages: int
    today_messages: int
    total_links: int


class DailyTrendItem(BaseModel):
    """每日趋势项"""
    date: str  # 格式：MM-DD
    messages: int
    links: int


class DailyTrendResponse(BaseModel):
    """最近10天趋势"""
    days: List[DailyTrendItem]


class DedupStatsItem(BaseModel):
    """去重统计项"""
    hour: datetime
    deleted_count: int


class DedupStatsResponse(BaseModel):
    """最近10小时去重统计"""
    hours: List[DedupStatsItem]


class NetdiskDistributionItem(BaseModel):
    """网盘分布项"""
    netdisk_name: str
    link_count: int
    percentage: float


class NetdiskDistributionResponse(BaseModel):
    """最近24小时网盘分布"""
    distribution: List[NetdiskDistributionItem]


class ActivityHeatmapCell(BaseModel):
    """活跃热力图单元格"""
    date: str  # 格式：YYYY-MM-DD
    hour: int  # 0-23
    message_count: int


class ActivityHeatmapResponse(BaseModel):
    """最近N天的活跃热力图"""
    dates: List[str]
    hours: List[int]
    cells: List[ActivityHeatmapCell]
    max_count: int

